core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerAdsLoader.java

```diff
@@ -171,14 +171,23 @@ public void onTimelineChanged(Timeline timeline, @Nullable Object manifest, int
         if (advertHasNotStarted()) {
             adGroupIndex = player.getCurrentAdGroupIndex();
             adIndexInGroup = player.getCurrentAdIndexInAdGroup();
-            notifyAdvertStart(advertBreaks.get(adGroupIndex));
+
+            if (canPlayAdverts(adGroupIndex)) {
```

Comment:
This checks the position information is correct within the `exo-player`, if it is and that particular advert has `AVAILABLE` adverts then we can play.